support-workers/src/typescript/util/zuoraDateReplacer.ts

```diff
@@ -1,14 +1,52 @@
 import { zuoraDateFormat } from '@modules/zuora/utils/common';
 import dayjs from 'dayjs';
 
-const isoDateRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z$/;
+/**
+ * Recursively clones an input value, replacing date instances
+ * with the formatted string from `zuoraDateFormat`.
+ * Other values are left unchanged. Arrays and plain objects are deep-copied.
+ * Does not support circular references.
+ */
+export type ReplaceDates<T> = T extends Date
+	? string
+	: T extends (...args: unknown[]) => unknown
+	? T
+	: T extends string
+	? string
+	: T extends Array<infer U>
+	? Array<ReplaceDates<U>>
+	: T extends object
+	? { [K in keyof T]: ReplaceDates<T[K]> }
+	: T;
 
-export function zuoraDateReplacer(key: string, value: unknown): unknown {
-	if (typeof value === 'string' && isoDateRegex.test(value)) {
-		const date = new Date(value);
-		if (!isNaN(date.getTime())) {
-			return zuoraDateFormat(dayjs(date));
+export function replaceDatesWithZuoraFormatDeep<T>(input: T): ReplaceDates<T> {
+	function formatIfDateLike(v: unknown): unknown {
+		if (v instanceof Date && !isNaN(v.getTime())) {
```

Comment:
Could you have an input type of `Date` instead of `unknown` and possibly loose the `instanceof` check?